ci/scripts/emscripten_python_build.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -ex
+
+source ~/emsdk/emsdk_env.sh
+
+source_dir=${1}/python
+cd ${source_dir}
+rm -rf dist
+pyodide build
```

Comment:
Could you use build directory instead of changing source directory?
It creates files owned by `root` on host.
See also: https://github.com/apache/arrow/issues/41429

`python_build.sh` copies sources for now: https://github.com/apache/arrow/blob/255dbf990c3d3e5fb1270a2a11efe0af2be195ab/ci/scripts/python_build.sh#L81-L93